modules/alb-controller/main.tf

```
@@ -2,23 +2,35 @@ locals {
   enabled = module.this.enabled
 }
 
+resource "kubernetes_namespace" "default" {
+  count = local.enabled && var.create_namespace ? 1 : 0
+
+  metadata {
+    name = var.kubernetes_namespace
+
+    annotations = {}
+
+    labels = module.this.tags
+  }
+}
+
 module "alb_controller" {
   source  = "cloudposse/helm-release/aws"
-  version = "0.2.1"
+  version = "0.5.0"
 
   name                 = "" # avoids hitting length restrictions on IAM Role names
   chart                = var.chart
   repository           = var.chart_repository
   description          = var.chart_description
   chart_version        = var.chart_version
-  kubernetes_namespace = var.kubernetes_namespace
-  create_namespace     = var.create_namespace
+  kubernetes_namespace = join("", kubernetes_namespace.default.*.id)
```

This line above assumes that the `var.create_namespace` will always be true.

We should solve for the case when it is false by passing in the `kubernetes_namespace` input directly

```suggestion
  kubernetes_namespace = local.enabled && var.create_namespace ? join("", kubernetes_namespace.default.*.id) : var.kubernetes_namespace
```